sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingSymmetricHashJoinHelper.scala

```diff
@@ -219,10 +222,35 @@ object StreamingSymmetricHashJoinHelper extends Logging {
           attributesWithEventWatermark = AttributeSet(otherSideInputAttributes),
           condition,
           eventTimeWatermarkForEviction)
-        val inputAttributeWithWatermark = oneSideInputAttributes.find(_.metadata.contains(delayKey))
```

Comment:
Where is the equivalent of this statement? This looks like the event-time attribute from one side (left side in examples).

Reply:
The code previously made the over-simplified assumption that there would be a watermark on one side, the state removal side. It would then use this assumption to find the attribute for state removal.

This assumption is not right. If you have the expression `L > R + 10` and `R` has a watermark, how do you find `L`? The approach I have chosen to take is taking the attributes (`attributesInCondition`), filtering for the attributes on the removal side (`oneSideInputAttributes`), and then taking that result. That would leave us with `R`.

Effectively, this line is equivalent to `oneSideStateWatermarkAttributes.head`.

Reply:
> filtering for the 

Did you mean filtering out? 

> Effectively, this line is equivalent to oneSideStateWatermarkAttributes.head.

`This line`: Is that the line you removed?

Reply:
> Did you mean filtering out?

Well, we _want_ the attributes on `oneSideInputAttributes`, so it's not filtering out, it's filtering for. If you look at the new `.filter` condition, we are doing `oneSideInputAttributeSet.contains(a)`, so we only get back the attributes that are in `oneSideInputAttributes`.